### PR TITLE
[PodVersion] Undelete pod on push when necessary

### LINKED ISSUE #172: "name is already taken" after a pod version delete
I've deleted a pod version with `pod trunk delete ASProgressHud.podspec 0.0.1`, now i'm trying to push the version 0.0.2,  but when i pushe again i've this error **[!] {"name"=>["is already taken"]}**

### PR DESCRIPTION
**Pull request by segiddins**
Closes #172.

\c @orta @floere @alloy

### CODE REVIEW

**Comment by orta** [comment]
This looks good to me :+1:

File: app/models/pod.rb [hunk 1]
```diff
@@ -15,10 +15,16 @@ class Pod < Sequel::Model
       one_to_many :versions, :class => 'Pod::TrunkApp::PodVersion'
       many_to_many :owners
 
+      alias_method :deleted?, :deleted
+
       # Finds a pod by name that has *not* been deleted.
```
Comment (alloy) [inline app/models/pod.rb]: This comment is no longer up-to-date.

File: app/models/pod.rb [hunk 2]
```diff
@@ -26,8 +32,8 @@ def self.find_by_name(name)
       # If the pod has owners but the specified user is not one of the owners the 'no access
       # allowed' block will be called.
       #
-      def self.find_by_name_and_owner(name, owner)
-        if pod = find_by_name(name)
+      def self.find_by_name_and_owner(name, owner, include_deleted: false)
```
Comment (alloy) [inline app/models/pod.rb]: Is this still being used elsewhere without `include_deleted => true` ?

File: app/models/pod_version.rb [hunk 3]
```diff
@@ -80,6 +80,7 @@ def push!(committer, specification_data, change_type)
           add_commit(:committer => committer, :sha => response.commit_sha, :specification_data => specification_data)
           pod.add_owner(committer) if pod.owners.empty?
         end
+        pod.update(:deleted => pod.versions_dataset.first(:deleted => false).nil?)
```
Comment (alloy) [inline app/models/pod_version.rb]: Maybe doing a `count` and comparing to `0` is a bit clearer.

**Comment by alloy** [comment]
Aside from those small comments this looks good to me :+1:

File: app/models/pod.rb (hunk 2 above)
Comment (segiddins) [inline app/models/pod.rb]: yup, its being used in a bunch of places in the pods controller

**Comment by segiddins** [comment]
@alloy comments addressed

File: app/models/pod.rb (hunk 2 above)
Comment (alloy) [inline app/models/pod.rb]: :+1:

**Comment by alloy** [comment]
Aaaccceee!

**Pull request merged**